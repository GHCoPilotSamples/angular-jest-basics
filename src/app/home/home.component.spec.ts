import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { capitalize } from 'lodash';
import { CapitalizePipe } from '../shared/capitalize.pipe';
import { HighlightDirective } from '../shared/highlight.directive';
import { HomeComponent } from './home.component';
import { HomeService } from './home.service';

const testData = [
  {
    name: 'trulli',
    image: 'pic_trulli.jpg',
    alt: 'Italian Trulli',
  },
  {
    name: 'chania',
    image: 'img_chania.jpg',
    alt: 'Chania',
  },
];
class HomeServiceStub {
  getCities() {
    return Promise.resolve(testData);
  }
}

describe('Home Component', () => {
  let fixture: ComponentFixture<HomeComponent>;
  let homeComponent: HomeComponent;
  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [HomeComponent, CapitalizePipe, HighlightDirective],
    }).overrideComponent(HomeComponent, {
      set: {
        providers: [
          {
            provide: HomeService,
            useClass: HomeServiceStub,
          },
        ],
      },
    });
    fixture = TestBed.createComponent(HomeComponent);
    homeComponent = fixture.componentInstance;
    fixture.detectChanges();
  });

  it(
    'is created and data received',
    waitForAsync(() => {
      fixture.whenStable().then(() => {
        expect(homeComponent).toBeDefined();
        expect(homeComponent.cities).toEqual(testData);
      });
    })
  );

 
