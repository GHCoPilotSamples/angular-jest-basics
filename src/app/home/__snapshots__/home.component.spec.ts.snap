// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component verify html 1`] = `
<app-home
  __ngContext__={[Function LRootView]}
  cities={[Function Array]}
  homeService={[Function HomeServiceStub]}
>
  <ul>
    <li>
      <p
        highlight="yellow"
        style="background-color: yellow;"
      >
        Trulli
      </p>
      <img
        alt="Italian Trulli"
        src="assets/pic_trulli.jpg"
      />
    </li>
    <li>
      <p
        highlight="yellow"
        style="background-color: yellow;"
      >
        Chania
      </p>
      <img
        alt="Chania"
        src="assets/img_chania.jpg"
      />
    </li>
  </ul>
</app-home>
`;
